modules/ssh/ssh.go

```diff
@@ -329,22 +331,42 @@ func FetchContentsOfFileE(t *testing.T, host Host, useSudo bool, filePath string
 	return CheckSshCommandE(t, host, command)
 }
 
-func listFileInRemoteDir(t *testing.T, sshSession *SshSession, options ScpDownloadOptions) ([]string, error) {
+func listFileInRemoteDir(t *testing.T, sshSession *SshSession, options ScpDownloadOptions, useSudo bool) ([]string, error) {
 	logger.Logf(t, "Running command %s on %s@%s", sshSession.Options.Command, sshSession.Options.Username, sshSession.Options.Address)
 
 	var result []string
 
 	findCommand := fmt.Sprintf("find %s -type f", options.RemoteDir)
 
-	if options.FileNameFilter != "" {
-		findCommand = fmt.Sprintf("%s -name %s", findCommand, options.FileNameFilter)
+	if useSudo {
+		findCommand = fmt.Sprintf("sudo %s", findCommand)
+	}
+
+	filtersLength := len(options.FileNameFilters)
+	if nil != options.FileNameFilters && filtersLength > 0 {
+
+		findCommand = fmt.Sprintf("%s \\(", findCommand)
+		for i, curFilter := range options.FileNameFilters {
+			findCommand = fmt.Sprintf("%s -name %s", findCommand, curFilter)
+
+			// only add the or flag if we're not the last element
+			if filtersLength-i > 1 {
+				findCommand = fmt.Sprintf("%s -o", findCommand)
+			}
```

Comment:
It seems like this isn't handling spaces between params correctly. A cleaner way to do this is, instead of string concatenation, add these args to a list, and then call `strings.Join` on the list at the end.

Reply:
Done.